src/workers/airtable-sync/worker.js

```diff
@@ -1,68 +1,54 @@
-  return () => {
-    console.log(`Stopping ${taskName}`);
-    running = false;
-  };
-}
 
 function startWorker(interval) {
-  if (interval < defaultInterval) {
+  let pollInterval = interval;
+  if (pollInterval < defaultInterval) {
     console.log(
-      `Interval ${interval} is too low. Clamping to ${defaultInterval}`
+      `Interval ${pollInterval} is too low. Clamping to ${defaultInterval}`
     );
-    interval = defaultInterval;
+    pollInterval = defaultInterval;
   }
-  let requestChanges = new ChangeDetector("Requests", 100);
-  schedule("airtable-sync.requests", interval, async () => {
-    const recordsChanged = await requestChanges.poll();
-    const statusFieldName = "Status";
-    const codeFieldName = "Code";
-    const slackIdFieldName = "Delivery slackid";
-    const triggerBackfillFieldName = "Trigger Backfill";
-    console.info(`Found ${recordsChanged.length} changes in Requests`);
-    for (const record of recordsChanged) {
-      //TODO: use this a few more times from different contexts and think about refactoring the api
-      if (record.didChange(statusFieldName)) {
-        const status = record.get(statusFieldName);
-        const newStatus = record.getPrior(statusFieldName);
-        console.log(
-          `${record.get(codeFieldName)} moved from ${newStatus} -> ${status}`
-        );
-      }
-      if(record.didChange(statusFieldName) || record.didChange(slackIdFieldName)){
-        await updateMessageContent(record)
-      }
-      if(record.didChange(triggerBackfillFieldName)){
-        await updateMessageContent(record)
-      }
-    }
+  const requestChanges = new ChangeDetector(airbase("Requests"), {
+    writeDelayMs: 100,
+    lastProcessedFieldName: "Last Processed",
+    sensitiveFields: SENSITIVE_FIELDS
   });
+  requestChanges.pollWithInterval(
+    "airtable-sync.requests",
+    interval,
+    async recordsChanged => {
+      const statusFieldName = "Status";
+      const codeFieldName = "Code";
+      const slackIdFieldName = "Delivery slackid";
+      const triggerBackfillFieldName = "Trigger Backfill";
+      console.info(`Found ${recordsChanged.length} changes in Requests`);
+      const promises = [];
+      recordsChanged.forEach(record => {
+        // TODO: use this a few more times from different contexts and think about refactoring the api
+        if (record.didChange(statusFieldName)) {
+          const status = record.get(statusFieldName);
+          const newStatus = record.getPrior(statusFieldName);
+          console.log(
+            `${record.get(codeFieldName)} moved from ${newStatus} -> ${status}`
+          );
+        }
+        if (
+          record.didChange(statusFieldName) ||
+          record.didChange(slackIdFieldName)
+        ) {
+          promises.push(updateMessageContent(record));
+        }
+        if (record.didChange(triggerBackfillFieldName)) {
+          promises.push(updateMessageContent(record));
+        }
+      });
+      return Promise.all(promises);
```

Comment:
Nice! The only downside I can think of is that we have to watch out for weird data races and are more likely to blow through Airtable's request limit (though we need to solve that more holistically anyhow).

Reply:
very true, I'll leave a note